spec/models/piece_spec.rb

```diff
@@ -50,4 +54,124 @@
       king.errors.clear
     end
   end
+
+  describe 'is_obstructed?' do
+    before :all do
+      # @user = FactoryGirl.create(:user)
+      # sign_in @user
+      @game = FactoryGirl.create(:game)
+      @game.pieces.delete
+      @white_queen = Queen.create(
+        x_position: 0,
+        y_position: 2,
+        color: true,
+        game_id: @game.id,
+        captured: false
+      )
+      @white_knight = Knight.create(
+        x_position: 0,
+        y_position: 1,
+        color: true,
+        game_id: @game.id,
+        captured: false
+      )
+      @white_pawn = Pawn.create(
+        x_position: 1,
+        y_position: 1,
+        color: true,
+        game_id: @game.id,
+        captured: false
+      )
+      @white_bishop = Bishop.create(
+        x_position: 4,
+        y_position: 2,
+        color: true,
+        game_id: @game.id,
+        captured: false
+      )
+      @white_rook = Rook.create(
+        x_position: 7,
+        y_position: 2,
+        color: true,
+        game_id: @game.id,
+        captured: false
+      )
+      @black_pawn = Pawn.create(
+        x_position: 4,
+        y_position: 6,
+        color: false,
+        game_id: @game.id,
+        captured: false
+      )
+      @black_queen = Queen.create(
+        x_position: 6,
+        y_position: 4,
+        color: false,
+        game_id: @game.id,
+        captured: false
+      )
+    end
+
+    it 'will be false when SE diag path is clear' do
+      expect(@white_queen.is_obstructed?(2, 4)).to be_falsey
+    end
+
+    it 'will be true when SW diag path has block, neg direction' do
+      expect(@black_queen.is_obstructed?(3, 7)).to eq true
+    end
+
+    it 'will be false when NW diag path is clear, piece at endpoint' do
+      expect(@black_queen.is_obstructed?(4, 2)).to eq false
+    end
+
+    it 'will be true when NW diag path blocked' do
+      expect(@black_queen.is_obstructed?(3, 1)).to be_truthy
```

Comment:
should explicitly test 'to eq true' here, otherwise it will pass for anything that is not nil or false....which could be bad!